packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.tsx

```diff
@@ -0,0 +1,352 @@
+import React, { useEffect, useImperativeHandle, useMemo, useState } from 'react'
+import OpenData from './OpenData'
+import { AclEditor } from '../AclEditor/AclEditor'
+import useUpdateAcl from '../AclEditor/useUpdateAcl'
+import {
+  getAccessTypeFromPermissionLevel,
+  PermissionLevel,
+  permissionLevelToLabel,
+} from '../../utils/PermissionLevelToAccessType'
+import {
+  ANONYMOUS_PRINCIPAL_ID,
+  AUTHENTICATED_PRINCIPAL_ID,
+  PUBLIC_PRINCIPAL_ID,
+} from '../../utils/SynapseConstants'
+import {
+  ALL_ENTITY_BUNDLE_FIELDS,
+  EntityType,
+  ResourceAccess,
+  UserProfile,
+} from '@sage-bionetworks/synapse-types'
+import {
+  useCreateEntityACL,
+  useDeleteEntityACL,
+  useGetEntityBenefactorACL,
+  useSuspenseGetCurrentUserProfile,
+  useSuspenseGetEntityBundle,
+  useUpdateEntityACL,
+} from '../../synapse-queries'
+import { Alert, Stack } from '@mui/material'
+import { InheritanceMessage } from './InheritanceMessage'
+import { CreateOrDeleteLocalSharingSettingsButton } from './CreateOrDeleteLocalSharingSettingsButton'
+import isEqualTreatArraysAsSets from '../../utils/functions/isEqualTreatArraysAsSets'
+import useNotifyNewACLUsers from './useNotifyNewACLUsers'
+import { BackendDestinationEnum, getEndpoint } from '../../utils/functions'
+import { getDisplayNameFromProfile } from '../../utils/functions/DisplayUtils'
+import { AclEditorSkeleton } from '../AclEditor/AclEditorSkeleton'
+
+const availablePermissionLevels: PermissionLevel[] = [
+  'CAN_VIEW',
+  'CAN_DOWNLOAD',
+  'CAN_EDIT',
+  'CAN_EDIT_DELETE',
+  'CAN_ADMINISTER',
+]
+
+export type EntityAclEditorProps = {
+  entityId: string
+  onCanSaveChange: (canSaveChanges: boolean) => void
+  onUpdateSuccess: () => void
+}
+
+export type EntityAclEditorHandle = {
+  save: () => void
+}
+
+function getSubject(entityName: string) {
+  return `${entityName} (shared on Synapse)`
+}
+
+function getBody(profile: UserProfile, entityId: string) {
+  return `${getDisplayNameFromProfile(
+    profile,
+  )} has shared an item with you on Synapse:\n${getEndpoint(
+    BackendDestinationEnum.PORTAL_ENDPOINT,
+  )}Synapse:${entityId}`
+}
+
+const EntityAclEditor = React.forwardRef(function EntityAclEditor(
+  props: EntityAclEditorProps,
+  ref: React.ForwardedRef<EntityAclEditorHandle>,
+) {
+  const { entityId, onCanSaveChange, onUpdateSuccess } = props
+
+  const { data: ownProfile } = useSuspenseGetCurrentUserProfile()
```

Comment:
I decided to use `Suspense` and `useSuspenseQuery` here. The wrapped `Suspense` component (which is an ACL editor skeleton component) will be shown until these successfully resolve, so we don't need to handle loading states for these queries--IMO removing that loading state logic is worth the trade off in a component like this that is already complex.

If any of the queries fail, the error will be propagated to the nearest parent error boundary.